backend/src/main/java/com/bakdata/conquery/commands/RecodeStoreCommand.java

```diff
@@ -0,0 +1,155 @@
+
+		log.info("Recoding Storage at `{} ({})` to `{} ({})`", inStoreDirectory, inLogSize, outStoreDirectory, outLogSize);
+
+		final File[] environments = inStoreDirectory.listFiles(File::isDirectory);
+
+		if(environments == null){
+			log.error("In Store is empty");
+			return;
+		}
+
+		for (File environment : environments) {
+			final File environmentDirectory = new File(outStoreDirectory, environment.getName());
+			environmentDirectory.mkdirs();
+
+			processEnvironment(environment, inLogSize, environmentDirectory, outLogSize);
+		}
+	}
+
+	private void processEnvironment(File inStoreDirectory, long inLogSize, File outStoreDirectory, long outLogSize) {
+		final jetbrains.exodus.env.Environment inEnvironment = Environments.newInstance(
+				inStoreDirectory,
+				new EnvironmentConfig().setLogFileSize(inLogSize)
+									   .setEnvIsReadonly(true)
+									   .setEnvCompactOnOpen(false)
+									   .setGcEnabled(false) // we dump first, then enable GC.
+		);
+
+		final EnvironmentConfig outConfig = new EnvironmentConfig().setLogFileSize(outLogSize)
+															.setGcEnabled(false);
+		final jetbrains.exodus.env.Environment outEnvironment = Environments.newInstance(
+				outStoreDirectory,
+				outConfig
+		);
+
+
+		final List<String> stores = inEnvironment.computeInReadonlyTransaction(inEnvironment::getAllStoreNames);
+
+		log.info("Environment {} contains {} Stores {}", inStoreDirectory, stores.size(), stores);
+		// Clear it before writing
+		outEnvironment.clear();
+
+		for (String store : stores) {
+			final Store inStore =
+					inEnvironment.computeInExclusiveTransaction(tx -> inEnvironment.openStore(store, StoreConfig.USE_EXISTING, tx, false));
+
+			if (inStore == null) {
+				log.error("{} does not exist, aborting.", store);
+				return;
+			}
+
+			final Store outStore =
+					outEnvironment.computeInExclusiveTransaction(tx -> outEnvironment.openStore(store, StoreConfig.WITHOUT_DUPLICATES_WITH_PREFIXING, tx, true));
+
+			if (outEnvironment.computeInReadonlyTransaction(outStore::count) > 0) {
+				log.warn("Store is not empty, it will be cleared.");
+			}
+
+			doRecode(inEnvironment, inStore, outEnvironment, outStore);
+			log.info("Done writing {}.", store);
+		}
+
+		outConfig.setGcEnabled(true);
+		outEnvironment.gc();
+
+		outEnvironment.close();
+		inEnvironment.close();
+	}
+
+	private void doRecode(jetbrains.exodus.env.Environment inEnvironment, Store inStore, jetbrains.exodus.env.Environment outEnvironment, Store outStore) {
+		final Transaction readTx = inEnvironment.beginReadonlyTransaction();
+		final Transaction writeTx = outEnvironment.beginExclusiveTransaction();
+
+		final long count = inStore.count(readTx);
+		long processed = 0;
+
+		log.info("Contains {} Entries", count);
+
+		final Cursor cursor = inStore.openCursor(readTx);
+
+		while (cursor.getNext()) {
+			outStore.putRight(writeTx, cursor.getKey(), cursor.getValue());
+			if (++processed % (1 + (count / 10)) == 0) {
```

Comment:
Would it make sense to extract `++processed % (1 + (count / 10))` into a variable? 
If not I'd ask: Why `10`? :-)